autometa/common/exceptions.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Copyright 2020 Ian J. Miller, Evan R. Rees, Kyle Wolf, Siddharth Uppal,
+Shaurya Chanana, Izaak Miller, Jason C. Kwan
+
+This file is part of Autometa.
+
+Autometa is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+
+File containing customized AutometaExceptions for more specific exception handling
+"""
+
+class AutometaException(Exception):
+    """docstring for AutometaException."""
+
+    def __init__(self, value):
+        self.value = value
+
+    issue_request = '''
+    An error was encountered!
+
+    Please help us fix your problem!
+
+    You may file an issue with us at https://github.com/KwanLab/Autometa/issues/new
+    '''
+    def __str__(self):
+        return f'{self.value}\n\n{issue_request}'
+
+class KmerFormatError(AutometaException):
+    """KmerFormatError exception class."""
+
+    def __init__(self, fpath):
+        super(AutometaException, self).__init__(fpath)
+        self.fpath = fpath
+
+    def __str__(self):
+        return f'{self.fpath} does not contain a \"contig\" column. '\
+        'Ensure the k-mer matrix was properly generated.'
+
+class KmerEmbeddingError(AutometaException):
+    """KmerEmbeddingError exception class."""
+
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return self.value
+
+class RecursiveDBSCANError(AutometaException):
+    """RecursiveDBSCANError exception class."""
```

Comment:
Same comment here. From the code I can't tell why this exists and there is no message for the user.

Reply:
A message is passed in to the exception. The base AutometaException wrapping functionality is not yet implemented, but the base AutometaException can act as the wrapper for emitting a message to submit a user issue based on the error encountered. If an an exception is encountered but an AutometaException is not emitted, a new issue request can be suggested to the user so we can account for any bugs of which we are not aware.

Reply:
The `RecursiveDBSCANError` is passed a message and is described here:

https://github.com/WiscEvan/Autometa/blob/5c735796764df761558d5035bfd7248409a5fe1a/autometa/binning/recursive_dbscan.py#L340-L343